autopilot/contractor.go

```diff
@@ -669,7 +678,8 @@ func (c *contractor) managedFindMinAllowedHostScores(ctx context.Context, cfg ap
 		return 0, err
 	}
 	if len(hosts) == 0 {
-		return 0, errors.New("no hosts returned in candidateHosts")
+		c.logger.Warn("min host score is 0 because there are no candidate hosts")
+		return 0, nil
```

Comment:
Can you update this to math.SmallestNonzeroFloat64? I think it makes sense to reserve 0 for hosts which are especially bad so we want to filter those out even if we can't determine a min score from the known hosts.